[ch35137] context campaign is missing from homepage

aliceathens opened the pull request.

aliceathens committed:
Enable all URL's to send the ctx.campaign

- By removing location.search condition (query) which was omitting the home page

aliceathens committed:
Make the same change to analytics.js.1

aliceathens committed:
Update analytics.js-integration-segmentio hash to latest changes

aliceathens committed:
Undo old changes

aliceathens commented on analytics.js:
```diff
@@ -2964,7 +2964,7 @@ module.exports={
   "_args": [
     [
       "@segment/analytics.js-core@3.2.5",
-      "/Users/jopdeklein/Development/analytics.js"
```
@jopdeklein not sure if this is an issue now its changed to my local env..?

jopdeklein replied:
👍  I never noticed it was pointing to a local path - since Gerrit also was able to build, I believe it's OK (as long as we have the .min version working fine I expect it to be OK :))

aliceathens commented on analytics.js:
```diff
@@ -5670,8 +5670,7 @@ function object(obj, strict) {
-      // obj[key] = isodate.parse(val);
-      obj[key] = val;
+      obj[key] = isodate.parse(val);
     } else if (type(val) === 'object' || type(val) === 'array') {
```
This change and the one below is related to this [PR](https://github.com/FindHotel/analytics.js/pull/2) I also see you changes analytics.js.1 do you know the purpose of this file? 
Let me know if i should make these manual changes as well as on line 5699 below?

jopdeklein replied:
I see, it looks like we previously skipped the step of updating the `analytics.js-integration-findhotel` and just applied them here, which means now they will get overridden when we rebuilt from your changes to the FH integration, sorry about that :( 

I think the best thing to do is to: 
1. re-apply these changes in `analytics.js-integration-findhotel` in a PR there, and merge them to master (with the commit: Prevent Date / Time manipulation for date-like strings)
1. refer to that commit in this package.json
1. rebuild this project - hopefully the diff will be just your original changes for the campaign context

Sorry for this mess, let me know if I can help.

aliceathens replied:
@jopdeklein makes sense however it seems that the changes applied do not come from `analytics.js-integration-findhotel` instead I traced them back to this
[repo](https://github.com/segmentio/isodate-traverse/blob/1.0.1/lib/index.js) 😬
I guess one good thing is that there was a good reason for you applying this here.

I guess we have a few options:
1. Fork the `isodate-traverse` repo from 1.0.1 and make the customisations on that but not sure which package includes it just now and how we can override it with our repo.
2. Do a manual patch again and document it for next time.
3. Try to update the `isodate-traverse` repo to the latest and see if that fixes what you were trying to do originally (risky option)

Let me know your thoughts I am feeling that option 2 might be best again for now?...

aliceathens commented on analytics.js:
```diff
@@ -6403,7 +6403,7 @@ var _clockseq;
 
-// See https://github.com/broofa/node-uuid for API details
+// See https://github.com/uuidjs/uuid for API details
 function v1(options, buf, offset) {
```
Not 100% sure why this comment changed...

jopdeklein replied:
That's probably due to the issue I mentioned above - I am assuming it was a package (lock) update in the FH integration, probably no issue :)

aliceathens commented:
@jopdeklein once we have approved this PR and I know the steps to deploy this package.. I will add updated steps to the readme but for now there are some outstanding comments above we should fix first! Thanks 😊

jopdeklein committed:
Add automated steps for applying Prevent Date / Time manipulation for date-like strings

jopdeklein committed:
Apply Prevent Date / Time manipulation for date-like strings fixes

aliceathens committed:
Remove analytics.js.1 as this is no longer needed

- it was to keep analytics.js “pure” and create a custom  .1 version by applying changes described in the readme but its no longer relevent
